spec/registry.md

```diff
@@ -255,3 +255,500 @@ which only expects the `accord` option:
+
+The function `:integer` provides a reduced set of options for selecting
+and formatting numeric values as integers.
+
+### Operands
+
+The _operand_ of a number function is either an implementation-defined type or
+a literal that matches the `number-literal` production in the [ABNF](/spec/message.abnf).
+All other values produce a _Resolution Error_.
+
+> For example, in Java, any subclass of `java.lang.Number` plus the primitive
+> types (`byte`, `short`, `int`, `long`, `float`, `double`, etc.) 
+> might be considered as the "implementation-defined numeric types".
+> Implementations in other programming languages would define different types
+> or classes according to their local needs.
+
+> [!NOTE]
+> String values passed as variables in the _formatting context_'s
+> _input mapping_ can be formatted as numeric values as long as their
+> contents match the `number-literal` production in the [ABNF](/main/spec/message.abnf).
+>
+> For example, if the value of the variable `num` were the string
+> `-1234.567`, it would behave identically to the local
+> variable in this example:
+> ```
+> .local $example = {|-1234.567| :number}
+> {{{$num :number} == {$example}}}
+> ```
+
+> [!NOTE]
+> Implementations are encouraged to provide support for compound types or data structures
+> that provide additional semantic meaning to the formatting of number-like values.
+> For example, in ICU4J, the type `com.ibm.icu.util.Measure` can be used to communicate
+> a value that include a unit
+> or the type `com.ibm.icu.util.CurrencyAmount` can be used to set the currency and related
+> options (such as the number of fraction digits).
+
+
+### Options
+
+The following options and their values are required in the default registry to be available on the 
+function `:number`:
+- `select`
+   -  `plural` (default)
+   -  `ordinal`
+   -  `exact`
+
+ 
+> [!IMPORTANT]
+> **Default Value of `select` Option**
+>
+> The value `plural` is default for the option `select` 
+> because it is the most common use case for numeric selection.
+> It can be used for exact value matches but also allows for the grammatical needs of other 
+> languages using CLDR's plural rules.
+> This might not be noticeable in the source language (particularly English), 
+> but can cause problems in target locales that the original developer is not considering.
+>
+> For example, a naive developer might use a special message for the value `1` without
+> considering other locale's need for a `one` plural:
+>
+>```
+> .match {$var}
+> 1   {{You have one last chance}}
+> one {{You have {$var} chance remaining}} // needed by languages such as Polish or Russian
+>                                          // such locales typically require other keywords
+>                                          // such as two, few, many, and so forth
+> *   {{You have {$var} chances remaining}}
+>```
+
+- `compactDisplay` (this option only has meaning when combined with the option `notation=compact`)
```

Comment:
We don't do that for all of the other values--and sometimes this is very deliberate. We might refer to MDN or 402 for the compact display gunk, though, since otherwise this gives implementations the "freedom" to implement `short` and `long` but not other forms of `notation=compact`.

What this really argues for is: we should excommunicate `compactDisplay` and `notation` from Tech Preview and consider it for RGI instead. JS and ICU can implement as an extension. (Scientific and engineering notation are also specialized)